handler/v1/search.go

```diff
@@ -0,0 +1,49 @@
+package v1
+
+import (
+	"GithubSearch/types/v1/api"
+	"github.com/flannel-dev-lab/cyclops/v2/response"
+	"net/http"
+)
+
+func (h *Handler) Search(w http.ResponseWriter, r *http.Request) {
+	searchTerm := r.URL.Query().Get("search_term")
+	user := r.URL.Query().Get("user")
+
+	ctx := r.Context()
+
+	if searchTerm == "" || user == "" {
+		response.ErrorResponse(http.StatusBadRequest, "search term or user is empty", w)
+		return
+	}
+
+	githubSearchResponse, err := h.GithubClient.Search(ctx, user, searchTerm)
+	if err != nil {
+		response.ErrorResponse(http.StatusInternalServerError, "something is wrong", w)
```

Comment:
You should have a more descriptive error here. You should also at least log the actual `err` that was returned
```suggestion
		fmt.Println(err) // but use your logger
		response.ErrorResponse(http.StatusInternalServerError, "error during github search", w)
```